scripts: twister: valgrind improvements

aunsbjerg opened the pull request:
A few improvements to twister's valgrind integration:

* warn if no valgrind executable is found
* log path to valgrind.log on failures during valgrind step
* add `--track-origins=yes` to valgrind options

Adding `--track-origins=yes` will make valgrind execution a bit slower but will provide much more valuable information on failures.

hakehuang reviewed (dismissed):
a genernal comments, the three commits can be merged into 1, as all of changes are in one PR.

aunsbjerg commented:
> a genernal comments, the three commits can be merged into 1, as all of changes are in one PR.

My thinking is that it is three distinct changes, and that not all of them might be something we want to merge. But I can squash the commits if that's the way to go, no problem.

Commit:
twister: add --track-origin to valgrind execution

generate full information when looking for uninitialized variables

Signed-off-by: Mikkel Jakobsen <mikkel.aunsbjerg@escolifesciences.com>

Commit:
twister: error if valgrind executable not found

if --enable-valgrind is selected but no valgrind executable can be
found in PATH, exit with an error

Signed-off-by: Mikkel Jakobsen <mikkel.aunsbjerg@escolifesciences.com>

Commit:
twister: fix valgrind failure handling

falures found during valgrind execution were treated as normal test
failures, so the logger would display a link to handler.log and not to
valgrind.log, leading to some confusion.

the return code is now handled in a correct sequence.

Signed-off-by: Mikkel Jakobsen <mikkel.aunsbjerg@escolifesciences.com>

aunsbjerg commented on scripts/pylib/twister/twisterlib.py:
```diff
@@ -554,13 +554,16 @@ def handle(self):
                        ] + command
             run_valgrind = True
+            else:
+                logger.warning("valgrind executable not found in PATH")
```
Good idea, moved executable checking to `twister` and added return on failure.
